tests/std/tests/P0784R7_library_support_for_more_constexpr_containers/test.cpp

```diff
@@ -272,7 +266,249 @@ constexpr void test_compiletime() {
     }
 }
 static_assert((test_compiletime(), true));
-#endif // _HAS_CXX20 && defined(__cpp_constexpr_dynamic_alloc)
+
+template <class T>
+struct A {
+    T value;
+
+    constexpr A() noexcept = default;
+    constexpr ~A()         = default;
+};
+
+template <class T>
+struct nontrivial_A {
+    T value;
+
+    constexpr nontrivial_A(T in = T{}) noexcept : value(in) {}
+    constexpr ~nontrivial_A() {}
+};
+
+constexpr void test_compiletime_destroy_variants() {

```

Comment:
The one difference that we have between the old `destroy` and the new version is that we utilize `destroy_at` which handles arrays special.

Could we add some test coverage for something that stores an array of ints?

Reply:
I believe there is already a test that covers the array path in `destroy_at` and `ranges::destroy_at` in here:
https://github.com/microsoft/STL/blob/cf86b05c5e86e8a91cc67a57e622c55d06414091/tests/std/tests/P0784R7_library_support_for_more_constexpr_containers/test.cpp#L205-L213